noir-projects/noir-protocol-circuits/crates/private-kernel-lib/src/private_call_data_validator.nr

```diff
@@ -218,29 +183,46 @@ impl PrivateCallDataValidator {
         );
     }
 
-    pub fn validate_against_call_request(self, request: CallRequest) {
+    pub fn validate_against_call_request(self, scoped_call_request: ScopedPrivateCallRequest) {
         let call_stack_item = self.data.call_stack_item;
+        let caller_contract_address = scoped_call_request.contract_address;
+        let request = scoped_call_request.call_request;
 
         assert_eq(
             request.hash, call_stack_item.hash(), "calculated private_call_hash does not match provided private_call_hash at the top of the call stack"
         );
 
         let call_context = call_stack_item.public_inputs.call_context;
+        let caller_context = request.caller_context;
+        let is_caller_hidden = caller_context.is_hidden();
 
         if call_context.is_delegate_call {
-            let caller_context = request.caller_context;
-            assert(!caller_context.is_empty(), "caller context cannot be empty for delegate calls");
+            assert(!is_caller_hidden, "caller context cannot be hidden for delegate calls");
             assert_eq(
                 call_context.msg_sender, caller_context.msg_sender, "call stack msg_sender does not match expected msg_sender for delegate calls"
             );
             assert_eq(
                 call_context.storage_contract_address, caller_context.storage_contract_address, "call stack storage address does not match expected contract address for delegate calls"
             );
         } else {
+            assert(is_caller_hidden, "caller context must be hidden for non-delegate calls");
```

Comment:
Adding this check to ensure that the caller won't be leaked by the private function if it's not a delegate call.